### PR TITLE
Added change parent task application service

### PR DESCRIPTION
**Pull request by benatespina**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #145 
| License       | MIT

### CODE REVIEW

File: src/Kreta/TaskManager/Application/Project/Task/ChangeParentTaskHandler.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        );
+        $changerId = MemberId::generate(
+            UserId::generate(
+                $command->changerId()
+            ),
+            $organization->id()
+        );
+
+        if (!$organization->isMember($changerId)) {
+            throw new TaskCreationNotAllowedException();
```
Comment (gorkalaucirica) [inline src/Kreta/TaskManager/Application/Project/Task/ChangeParentTaskHandler.php]: Use `UnauthorizedTaskActionException`

File: src/Kreta/TaskManager/Application/Project/Task/ChangeParentTaskHandler.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            TaskId::generate(
+                $command->id()
+            )
+        );
+        if (!$task instanceof Task) {
+            throw new TaskDoesNotExistException();
+        }
+
+        if (null !== $parentId = $command->parentId()) {
+            if ($parentId === $task->id()->id()) {
```
Comment (gorkalaucirica) [inline src/Kreta/TaskManager/Application/Project/Task/ChangeParentTaskHandler.php]: Maybe we need `$task->equals($taskId)` or `$task->id()->equals($parentId)`

File: src/Kreta/TaskManager/Tests/Spec/Kreta/TaskManager/Domain/Model/Project/Task/TaskSpec.php [hunk 3]
```diff
@@ -113,6 +114,30 @@ function it_can_be_edited(TaskTitle $title)
         $this->shouldHavePublished(TaskEdited::class);
     }
 
+    function it_parent_can_be_changed(TaskId $parentId)
```
Comment (gorkalaucirica) [inline src/Kreta/TaskManager/Tests/Spec/Kreta/TaskManager/Domain/Model/Project/Task/TaskSpec.php]: `its_parent_can_be_changed`

File: src/Kreta/TaskManager/Tests/Spec/Kreta/TaskManager/Domain/Model/Project/Task/TaskSpec.php [hunk 4]
```diff
@@ -113,6 +114,30 @@ function it_can_be_edited(TaskTitle $title)
...
+
+        $this->changeParent($parentId);
+
+        $this->parentId()->shouldReturn($parentId);
+        $this->updatedOn()->shouldNotEqual($oldUpdatedOn);
+
+        $this->shouldHavePublished(TaskParentChanged::class);
+    }
+
+    function it_parent_can_be_changed_without_parent()
```
Comment (gorkalaucirica) [inline src/Kreta/TaskManager/Tests/Spec/Kreta/TaskManager/Domain/Model/Project/Task/TaskSpec.php]: `it_can_be_changed_to_root_task` ?

Comment (benatespina) [inline src/Kreta/TaskManager/Tests/Spec/Kreta/TaskManager/Domain/Model/Project/Task/TaskSpec.php]: I like `its_parent_can_be_removed`

**Pull request merged**